src/app/backend/handler/apihandler.go

```diff
@@ -173,13 +177,60 @@ func FormatResponseLog(resp *restful.Response, req *restful.Request) string {
 		req.Request.RemoteAddr, resp.StatusCode())
 }
 
-// CreateHTTPAPIHandler creates a new HTTP handler that handles all requests to the API of the backend.
-func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.HeapsterClient,
-	clientConfig clientcmd.ClientConfig) (http.Handler, error) {
+func (apiHandler *APIHandler) injectApiClient(fn func(*clientK8s.Clientset, *restful.Request, *restful.Response)) restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		apiclient, _, err := apiHandler.getApiClient(request)
+		if err != nil {
+			handleInternalError(response, err)
+			return
+		}
+
+		fn(apiclient, request, response)
+	}
+}
+
+func (apiHandler *APIHandler) getApiClient(request *restful.Request) (*clientK8s.Clientset, clientcmd.ClientConfig, error) {
+	authorizationHeader := request.HeaderParameter("Authorization")
+	token := ""
+	if strings.HasPrefix(authorizationHeader, "Bearer ") {
+		token = strings.TrimPrefix(authorizationHeader, "Bearer ")
+	}
+
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ ExplicitPath: apiHandler.clientConfig.KubeConfigFile},
+		&clientcmd.ConfigOverrides{
+			AuthInfo:clientcmdapi.AuthInfo{Token: token},
+			ClusterInfo: clientcmdapi.Cluster{Server: apiHandler.clientConfig.ApiserverHost}})
+
+	cfg, err := clientConfig.ClientConfig()
+
+	// TODO Workaround, because ClientConfig doesn't override the token from the tokenfile with the one specified
+	// when run inside a cluster
+	if token != "" {
+		cfg.BearerToken = token
```

Comment:
We should move the error check up before this so there isn't a panic when `cfg == nil`. I just hit this attempting to rebase on master.